src/atomdb/redis_mongodb/RedisContext.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <hiredis/hiredis.h>
+#include <hiredis_cluster/hircluster.h>
+
+#include <string>
+
+using namespace std;
+
+// Wrapper class for Redis context to handle both single and cluster modes
+class RedisContext {
+   public:
+    explicit RedisContext(bool is_cluster);
+    ~RedisContext();
+
+    void setContext(redisContext* ctx);
+    void setContext(redisClusterContext* ctx);
+
+    redisReply* execute(const char* command);
+    bool hasError() const;
+    const char* getError() const;
```

Comment:
The pattern used in this project for method names is _snake case_:

```suggestion
    void set_context(redisContext* ctx);
    void set_context(redisClusterContext* ctx);

    redisReply* execute(const char* command);
    bool has_error() const;
    const char* get_error() const;
```